src/xtgeo/common/sys.py

```diff
@@ -380,133 +415,136 @@ def splitext(self, lower=False):
+                # Write stream to a temporary file
+                fds, self._tmpfile = mkstemp(prefix="tmpxtgeoio")
+                os.close(fds)
+                with open(self._tmpfile, "wb") as newfile:
+                    newfile.write(self.file.getvalue())
 
-        elif (
-            isinstance(self._file, io.BytesIO)
-            and self._mode == "rb"
-            and not islinux  # Windows or Darwin
-        ):
-            # windows/mac miss fmemopen; write buffer to a tmp instead as workaround
-            fds, self._tmpfile = mkstemp(prefix="tmpxtgeoio")
-            os.close(fds)
-            with open(self._tmpfile, "wb") as newfile:
-                newfile.write(self._file.getvalue())
-
-        else:
-            fobj = self.name
-
-        if self._memstream:
+        if self.memstream:
             if islinux:
                 cfhandle = _cxtgeo.xtg_fopen_bytestream(fobj, self._mode)
             else:
                 cfhandle = _cxtgeo.xtg_fopen(self._tmpfile, self._mode)
-
         else:
             try:
                 cfhandle = _cxtgeo.xtg_fopen(fobj, self._mode)
             except TypeError as err:
-                raise IOError(f"Cannot open file: {fobj}") from err
+                raise IOError(f"Cannot open file: {fobj!r}") from err
 
         self._cfhandle = cfhandle
         self._cfhandlecount = 1
 
-        logger.info("Get initial SWIG C fhandle no %s", self._cfhandlecount)
+        logger.debug("Get initial SWIG C fhandle no %s", self._cfhandlecount)
         return self._cfhandle
 
-    def cfclose(self, strict=True):
-        """Close SWIG C file handle by keeping track of _cfhandlecount.
+    def cfclose(self, strict: bool = True) -> bool:
+        """
+        Close SWIG C file handle by keeping track of _cfhandlecount.
+
+        Returns:
+            True if cfhandle is closed.
 
-        Return True if cfhandle is really closed.
         """
-        logger.info("Request for closing SWIG fhandle no: %s", self._cfhandlecount)
+        logger.debug("Request for closing SWIG fhandle no: %s", self._cfhandlecount)
 
-        if self._cfhandle is None or self._cfhandlecount == 0:
+        if self._cfhandle == 0 or self._cfhandlecount == 0:
             if strict:
                 raise RuntimeError("Ask to close a nonexisting C file handle")
 
-            self._cfhandle = None
+            self._cfhandle = 0
             self._cfhandlecount = 0
             return True
 
-        if self._cfhandlecount > 1 or self._cfhandlecount == 0:
+        if self._cfhandlecount > 1:
             self._cfhandlecount -= 1
-            logger.info(
+            logger.debug(
                 "Remaining SWIG cfhandles: %s, do not close...", self._cfhandlecount
             )
             return False
 
-        if self._memstream and self._cfhandle and "w" in self._mode:
+        if (
+            isinstance(self.file, io.BytesIO)
+            and self._cfhandle > 0
```

Comment:
If `io.StringIO` is allowed here the write further down will break (writing `buf = bytes(npos)`, bytes, to a string buffer). I'm a bit surprised this hasn't broken in the past, but I could be missing something, or StringIO could just be quite rare (or rare in write mode).